openrouteservice/src/main/java/heigit/ors/routing/graphhopper/extensions/core/PrepareCoreLandmarks.java

```diff
@@ -0,0 +1,223 @@
+        if (algo instanceof AStar) {
+            if (!lms.isInitialized())
+                throw new IllegalStateException("Initalize landmark storage before creating algorithms");
+
+            double epsilon = opts.getHints().getDouble(Parameters.Algorithms.ASTAR + ".epsilon", 1);
+            AStar astar = (AStar) algo;
+
+            astar.setApproximation(
+                    new CoreLMApproximator(qGraph, this.graph.getNodes(), lms, activeLM, lms.getFactor(), false)
+                            .setEpsilon(epsilon));
+            return algo;
+        } else if (algo instanceof AStarBidirection) {
+            if (!lms.isInitialized())
+                throw new IllegalStateException("Initalize landmark storage before creating algorithms");
+
+            double epsilon = opts.getHints().getDouble(Parameters.Algorithms.ASTAR_BI + ".epsilon", 1);
+            AStarBidirection astarbi = (AStarBidirection) algo;
+
+            astarbi.setApproximation(
+                    new CoreLMApproximator(qGraph, this.graph.getNodes(), lms, activeLM, lms.getFactor(), false)
+                            .setEpsilon(epsilon));
+            return algo;
+        } else if (algo instanceof AlternativeRoute) {
+            if (!lms.isInitialized())
+                throw new IllegalStateException("Initalize landmark storage before creating algorithms");
+
+            double epsilon = opts.getHints().getDouble(Parameters.Algorithms.ASTAR_BI + ".epsilon", 1);
+            AlternativeRoute altRoute = (AlternativeRoute) algo;
+            //TODO  //TODO Should work with standard LMApproximator
+
+            altRoute.setApproximation(
+                    new CoreLMApproximator(qGraph, this.graph.getNodes(), lms, activeLM, lms.getFactor(), false)
+                            .setEpsilon(epsilon));
+            // landmark algorithm follows good compromise between fast response and exploring 'interesting' paths so we
+            // can decrease this exploration factor further (1->dijkstra, 0.8->bidir. A*)
+            altRoute.setMaxExplorationFactor(0.6);
+        }
+
+        return algo;
+    }
+
+    public boolean matchesFilter(PMap pmap){
+        //Returns true if the landmarkset is for the avoidables.
+        //Also returns true if the query has no avoidables and the set has no avoidables
+            if(landmarksFilter.isFilter(pmap)){
+                //printLandmarksLongLat();
+                return true;
+            }
+            return false;
+    }
+
+    /**
+     * This method is for debugging
+     */
+
+    public void printLandmarksLongLat(){
+        int[] currentSubnetwork;
+        for(int subnetworkId = 1; subnetworkId < lms.getSubnetworksWithLandmarks(); subnetworkId++){
+            System.out.println("Subnetwork " + subnetworkId);
+            currentSubnetwork = lms.getLandmarks(subnetworkId);
+            for(int landmark : currentSubnetwork){
+                System.out.println("[" + graph.getNodeAccess().getLon(landmark)
```

Comment:
Use logger rather than printline. This method might not be getting used anyway in which case it should be deleted